### ISSUE TITLE
bug(core): can't define property _: object is not extensible

### ISSUE THREAD

**Issue by bcgilliom** [prescottprue/redux-firestore#240]
**What is the current behavior?**
- creating the firestore store fails when using the latest `react-native-firebase` (`6.0.1`)

**What is the expected behavior?**
- probably shouldn't add behavior to the firebase object?

**Which version of redux-firestore are you using? What about other dependencies?**
```
 "@react-native-firebase/app": "^6.0.1",
    "@react-native-firebase/auth": "^6.0.1",
    "@react-native-firebase/firestore": "^6.0.1",
    "@react-native-firebase/messaging": "^6.0.1",
    "@react-native-firebase/storage": "^6.0.1",
    "react-redux-firebase": "next",
    "redux-firestore": "0.9.0",
```

**Which environments/browsers are affected by this issue? Did this work in previous versions or setups?**
react native

**Minimal demo to reproduce issue (using [codesandbox](https://codesandbox.io/) or similar)**
I will see if I can get to this...

********

The error I'm seeing is coming from `createFirestoreInstance` and seems to be when it assigns the helpers:

```
 firebase._ = merge(defaultInternals, firebase._); // eslint-disable-line no-param-reassign
```

It looks like rn-firebase has used `Object.freeze` (https://github.com/invertase/react-native-firebase/blob/2fed4a9b38935b1973ce05d8e472a51daf3e4e89/packages/app/lib/internal/registry/namespace.js#L251) on the firebase object. Has anyone go this working with rn-firebase 6?

**Comment by prescottprue** [comment]
This is intentional and is how react-redux-firebase + redux-firestore extends the Firebase SDK (which allows for this).

Not sure what the reason would be on the side of react-native-firebase to not allow for extending, most likely there will need to be a pretty big change to handle things differently.

Did you try creating a copy of react-native-firebase before passing it in? Using spread or Object.assign?

**Comment by bcgilliom** [comment]
I haven’t yet - just wanted to raise it just in case someone else had encountered this. It seems like cloning and then adding the helpers would work, but is that something that just needs to happen on init? I guess if the firebase object doesn’t change, that would be ok... I’m not too familiar with the internals of that. I will see what happens and report back!

**Comment by bcgilliom** [comment]
So far using spread and R.clone (ramda) haven't worked - it gets past the helpers being added, but it doesn't seem to get the whole object (original on the left):

<img width="900" alt="Screen Shot 2019-10-13 at 12 42 09 AM" src="https://user-images.githubusercontent.com/5893743/66711048-4c9d7700-ed52-11e9-9752-f29fb2046217.png">

Any thought on the best way forward?

**Comment by prescottprue** [comment]
@bcgilliom Makes sense, seems like the getters are not being copied.

I think it would be a good idea to file an issue on the react-native-firebase repo to see if there is any way for exporting extension of their objects since it is supported by Firebase. Hopefully they could make extending an option?

Firebase also had/has a way of extending their object (I believe now called `extendNamespace ` or `deepExtend`) - so wonder if the react-native-firebase would be interested in supporting that route  of extending the object.

This was a design decision quite a long time ago since folks wanted access to other Firebase statics and such - in future versions though this will most likely change.

**Comment by cristian-milea** [comment]
any chance for a hotfix on this issue?

**Comment by bcgilliom** [comment]
https://github.com/prescottprue/redux-firestore/pull/248 is what I'm using. The PR would need tests to be fixed if it's in the right direction

**Comment by cristian-milea** [comment]
@bcgilliom , thanks. I tried using your forks directly and I still get the same error
`"react-redux-firebase": "https://github.com/bcgilliom/react-redux-firebase.git#master",
   "redux-firestore": "https://github.com/bcgilliom/redux-firestore.git#fixes_for_rnfb6",
    `

**Comment by bcgilliom** [comment]
The firebase changes are in a branch too - just rebased on latest of master and squashed here https://github.com/bcgilliom/react-redux-firebase.git#fixes_for_rnfb6

**Comment by cristian-milea** [comment]
Thanks, @bcgilliom ! I really appreciate your help!

I now get the following message:

> TypeError: firebase.extendApp is not a function
> 
> This error is located at:
>     in ReactReduxFirebaseProvider (at App.js:28)

I tried `pod clean && pod install`, also `yarn start -- --reset-cache` but couldn't get past this. Any ideas?

**Comment by bcgilliom** [comment]
Ohhh, yep! I don't have it in front of me, but tou need to send `firebase: firebase.app()` (or similar) in the provider config

**Comment by cristian-milea** [comment]
it works! you saved my day!

in my store config:
`export const rrfProps = {
  firebase: firebase.app(),
  config: rrfConfig,
  dispatch: store.dispatch,
  createFirestoreInstance
}`
and then 
`<ReactReduxFirebaseProvider {...rrfProps}><App/></ReactReduxFirebaseProvider>`

**Comment by prescottprue** [comment]
@bcgilliom Amazing work here, thanks for looking into the `extendApp` route - we should be able to get this in the next major version for sure. Wondering if it is good to still have the full firebase instance passed then use `firebase.app().extendApp()` under the hood - open to thoughts on that

**Comment by bcgilliom** [comment]
@prescottprue That would be a little cleaner probably - I'm happy to try it the next time in there - probably when looking at fixing the tests for the PR... So, would `firebaseInstance` also be `firebase.app()` then? I guess we'd need to look at where extend app puts stuff?

**Comment by prescottprue** [comment]
As mentioned in [777 of react-redux-firebase](https://github.com/prescottprue/react-redux-firebase/issues/777), this should be fixed by updating to v6.3.0 of react-native-firebase. For anyone wondering,  [here is the commit which removes usage of `Object.freeze`](https://github.com/invertase/react-native-firebase/pull/3157/files#diff-55efcaa4f5f3cd77d42b7201a43349d6L251).

Going to close this since updating should make this no longer an issue. If anyone updates to v6.3.0 of react-native-firebase and this is still and issue, please reach out.

**Comment by bcgilliom** [comment]
@prescottprue glad it worked out! sorry if the extendApp Pr caused problems 😬 !
